test/conformance/kubetestgen/main.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/go-openapi/analysis"
+	"github.com/go-openapi/loads"
+	"github.com/go-openapi/spec"
+	"gopkg.in/yaml.v2"
+)
+
+type options struct {
+	schemaPath   string
+	match        string
+	area         string
+	behaviorsDir string
+}
+
+func parseFlags() *options {
+	o := &options{}
+	flag.StringVar(&o.schemaPath, "schema", "", "Path to the OpenAPI schema")
+	flag.StringVar(&o.match, "match", ".*", "Resource name must match this regexp")
+	flag.StringVar(&o.area, "area", "default", "Area name to use")
+	flag.StringVar(&o.behaviorsDir, "dir", "../behaviors/generated/", "Path to the behaviors directory")
+	flag.Parse()
+	return o
+}
+
+var defMap map[string]analysis.SchemaRef
+
+func main() {
+	defMap = make(map[string]analysis.SchemaRef)
+	o := parseFlags()
+	pattern, err := regexp.Compile(o.match)
+	if err != nil {
+		fmt.Printf("ERROR: %s\n", err.Error())
+		os.Exit(1)
+	}
+
+	d, err := loads.JSONSpec(o.schemaPath)
+	if err != nil {
+		fmt.Printf("ERROR: %s\n", err.Error())
+		os.Exit(1)
+	}
+	defs := d.Analyzer.AllDefinitions()
+	sort.Slice(defs, func(i, j int) bool { return defs[i].Name < defs[j].Name })
+
+	for _, d := range defs {
+		if !d.TopLevel {
+			continue
+		}
+		defMap[d.Ref.String()] = d
+	}
+
+	var suites []*Suite
+	for _, v := range defs {
+		if !v.TopLevel || !pattern.MatchString(v.Name) {
+			continue
+		}
+		name := trimObjectName(v.Name)
+
+		// Print the schema here if necessary
+		//printSchema("", name, v.Name, v.Schema)
+
+		suite := (SuiteForSchema(o.area, name, v.Schema))
+		suites = append(suites, suite)
+	}
+	countFields(suites)
+	printYAML(o.behaviorsDir+o.area, suites)
+}
+
+func printYAML(area string, suites []*Suite) {
+	fname := area + ".yaml"
+	f, err := os.Create(fname)
+	if err != nil {
+		fmt.Println("ERROR: %s\n", err.Error())
+	}
+	defer f.Close()
+	for _, suite := range suites {
+		y, err := yaml.Marshal(suite)
+		if err != nil {
+			fmt.Println("ERROR: %s\n", err.Error())
+			os.Exit(1)
+		}
+		f.WriteString(string(y))
+	}
+}
+
+func countFields(suites []*Suite) {
+	var fieldsMapping map[string]int
+	fieldsMapping = make(map[string]int)
+	for _, suite := range suites {
+		for _, behavior := range suite.Behaviors {
+			if _, exists := fieldsMapping[behavior.ApiType]; exists {
+				fieldsMapping[behavior.ApiType] += 1
+			} else {
+				fieldsMapping[behavior.ApiType] = 1
+			}
+		}
+	}
+	for k, v := range fieldsMapping {
+		fmt.Printf("Type %v, Count %v\n", k, v)
```

Comment:
If we're going to print these out they should be comments

Reply:
We're writing the actual generated behaviours to a yaml file. This output acts as a summary and is written to stdout. What did you mean by comments?